src/app/model/Rock.ts

```diff
@@ -0,0 +1,109 @@
+import { Board } from './Board';
+import { PieceMovement } from './PieceMovement';
+import { Side, Square } from './Types';
+
+export class Rock implements PieceMovement {
+  id: string;
+  side: Side;
+
+  constructor(id: string, side: Side) {
+    this.id = id;
+    this.side = side;
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+    movesToGo.concat(this._goUp(position, board));
```

Comment:
Możesz mi opisać jak działa concat :) ?

Reply:
no dobra to trochę faktycznie jest źle

Reply:
Testy i tak najprawdziwsza prawde Ci powiedza :) Jak nie wylapie tego moj "naoczny" kompilator wczesniej :)